failsafe-controller/src/main/java/net/jodah/failsafe/package-info.java

```diff
@@ -0,0 +1,445 @@
+ * asynchronous nature of this design which would then allow the tester to mock, control, or script
+ * its execution using the Failsafe controller and thus more easily control the sequence of events.
+ *
+ * <p>The Failsafe controller provides a very simple synchronization mechanism between the test code
+ * and what happens in Failsafe using a notion of conditions or latches that can be notified and
+ * waited upon from either the mocked task side or from the test code side based on the testing
+ * need. For example, one could have Failsafe respond in error to an attempt until such a condition
+ * is notified by the test code when it is time to have the asynchronous code return successfully.
+ * Or again the asynchronous task could notify the test code it has completed a certain stage thus
+ * unblocking the test code from continuing with additional verification.
+ *
+ * <p>It is also possible to have the asynchronous task either block or wait for the whole operation
+ * to be canceled and for the test code to also wait for a given execution to complete successfully
+ * or not before proceeding.
+ *
+ * <p><b>Retry Policies</b>
+ *
+ * <p>The Failsafe controller will uses the provided retry policy as is except for the following
+ * exceptions:
+ *
+ * <ul>
+ *   <li>Time is compress such that it doesn't wait before handling the next attempt (unless
+ *       specifically mocked/scripted using <i>wait</i> actions)
+ *   <li>It will automatically abort whenever an {@link java.lang.AssertionError} is detected such
+ *       that the test would abort in failure.
+ * </ul>
+ *
+ * <p><b>Scheduled Executor</b>
+ *
+ * <p>The Failsafe controller will uses the provided scheduled executor as is. It does intercept all
+ * request for execution to this executor in order to determine when a particular Failsafe execution
+ * has completed including dispatching all asynchronous events that are done using this executor.
+ *
+ * <p><b>Injection</b>
+ *
+ * <p>The production code has to be modified such that it becomes possible to inject the Failsafe
+ * controller in lieu of Failsafe. To better understand how this is done, let's look at how one
+ * normally invokes Failsafe with an example:
+ *
+ * <p><code><pre>
+ *     Future&lt;MyObject&gt; future = Failsafe.with(myRetryPolicy)
+ *       .with(myScheduledExecutor)
+ *       .onRetry(this::logFailure)
+ *       .onAbort(this::logInterruptionAndRecreate)
+ *       .onFailure(this::logAndRecreateIfNotCancelled)
+ *       .onSuccess(this::logAndSetCreated)
+ *       .get(this::create);
+ * </pre></code>
+ *
+ * <p>In this example, the first line will create a {@link net.jodah.failsafe.SyncFailsafe} object.
+ * The second line will create a {@link net.jodah.failsafe.AsyncFailsafe} object from it and the
+ * next 4 lines registers event listeners. Additional lines could be added to further configure
+ * Failsafe if required. The final line is the one that actually triggers the start of an execution
```

Comment:
✏️ Consider changing this last sentence to:
"The final line is the one that triggers the start of an execution. It returns a {@link java.util.concurrent.Future} allowing access to the result at the end and also provides the ability to cancel the execution."